src/thumbbar.py

```diff
@@ -124,9 +119,9 @@ def _load(self):
         else:
             create = prefs['create thumbnails']
         self._stop_update = False
-        for i in xrange(1, self._window.file_handler.get_number_of_pages() + 1):
+        for i in range(1, self._window.file_handler.get_number_of_pages() + 1):
```

Comment:
How do you want to reduce the memory imprint of range() for py27 ?
I understand the list may very rarely be huge, but this was a good practice to use xranged in py27 (which replaced range in py3x)

Reply:
Most of the xrange/range issues had to do with very small lists of numbers (<300)  You are right that this could get very large depending on how it is used.   I've added some compatibility code at the top of files where the range can be variable

Reply:
Ok, if only for my education, what's the compatibility bit you've added in those files ?

Reply:
```python
try:
     range = xrange  # Python2
 except NameError:
     pass
```

In Python 2 it reassigns the name `range` to `xrange`.   In Python3 it gets a NameError since there is no `xrange` and so `range` stays `range`  This works because in Python3 `range` IS `xrange` so the code makes the two environments function the same.